src/palette/internal/paletteprovider.cpp

```diff
@@ -617,6 +616,22 @@ void PaletteProvider::init()
     });
 }
 
+void PaletteProvider::setFilter(const QString& filter)
+{
+    // Remove the model when there is no search text to return no results
+    // and thus speed up the opening of the palette search text box.
+    // Restore the model as soon as the search text is non-empty.
+    if (filter.length() > 0) {
+        m_searchFilterModel->setFilterFixedString(filter);
+        if (m_searchFilterModel->sourceModel() == nullptr) {
+            m_searchFilterModel->setSourceModel(m_masterPaletteModel);
+        }
+    } else {
+        m_searchFilterModel->setSourceModel(nullptr);
+        m_searchFilterModel->setFilterFixedString(filter);
```

Comment:
This is slightly clearer:

`m_searchFilterModel->setFilterFixedString(QString());`

Reply:
Thanks, guys! Should I amend my commit or create a new one with your suggestions?

Reply:
Since these changes are relatively simple, amending the commit is fine in this case!

Reply:
Done.